### ISSUE TITLE
Allow to pass features to cargo install in rust hooks

### ISSUE THREAD

**Issue by Terseus** [pre-commit/pre-commit#3230]
### search you tried in the issue tracker

rust featurres

### describe your actual problem

The rust hooks execute `cargo install --bins --root <root-dir> <packages>` to install the hook, however if a hook wants to use a different set of features than its default features it can't because there's no way to pass either additional flags to the install command or a `--features <features>` to cargo command.

See https://github.com/JohnnyMorganz/StyLua/issues/867 for a real world example where this has become a limitation.

It would be great to have a way to configure a hook to pass cargo additional features when building the hook's package.

### pre-commit --version

3.7.1

**Comment by asottile** [comment]
I suspect this should just work through `additional_dependencies`, similar to how it's used for python

**Comment by Terseus** [comment]
@asottile 
> I suspect this should just work through `additional_dependencies`, similar to how it's used for python

Right now in rust that option [is already in use](https://pre-commit.com/#rust):
> When specifying [additional_dependencies](https://pre-commit.com/#config-additional_dependencies) for Rust, you can use the syntax {package_name}:{package_version} to specify a new library dependency (used to build your hook repo), or the special syntax cli:{package_name}:{package_version} for a CLI dependency (built separately, with binaries made available for use by hooks).

Maybe we could add a new syntax for `additional_dependencies` in that e.g. values like `feature-name` could be passed as `--features name`.

**Comment by asottile** [comment]
yes I mean I suspect we can just send any arguments along to cargo install that don't fit the current format
